src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -27,9 +27,14 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
     /// </summary>
     public class DefaultRoslynCompilationService : ICompilationService
     {
-        private readonly DebugInformationFormat _pdbFormat = SymbolsUtility.SupportsFullPdbGeneration() ?
-            DebugInformationFormat.Pdb :
+        private readonly DebugInformationFormat _pdbFormat =
+#if NET451
+            SymbolsUtility.SupportsFullPdbGeneration() ?
+                DebugInformationFormat.Pdb :
+                DebugInformationFormat.PortablePdb;
+#else
             DebugInformationFormat.PortablePdb;
+#endif
```

Comment:
Why two `#if`s for one issue? Seems simpler to have one bit of conditional compilation in `SymbolsUtility.SupportsFullPdbGeneration()` -- return `false` when `!NET451`.